maven_iuvs/graphics/line_fit_plot.py

```diff
@@ -50,72 +75,100 @@ def detector_image(myfits, integration=0,
 
     cmap = idl_colorbars.getcmap(cmap)
 
-    ax.set_xlim([0, 1024])
-    ax.set_ylim([0, 1024])
-
-    # get the data
-    data = myfits['detector_dark_subtracted'].data[integration]
-
-    # figure out the binning
-    spapixlo = myfits['Binning'].data['SPAPIXLO'][0]
-    spapixhi = myfits['Binning'].data['SPAPIXHI'][0]
-    spepixlo = myfits['Binning'].data['SPEPIXLO'][0]
-    spepixhi = myfits['Binning'].data['SPEPIXHI'][0]
-    if not (set((spapixhi[:-1]+1)-spapixlo[1:]) == {0}
-            and set((spepixhi[:-1]+1)-spepixlo[1:]) == {0}):
-        raise ValueError("Error in bin table")
-
-    spepixrange = np.concatenate([[spepixlo[0]], spepixhi+1])
-    spapixrange = np.concatenate([[spapixlo[0]], spapixhi+1])
+    # ax.set_xlim([0, 1024])
+    # ax.set_ylim([0, 1024])
+
+    # This section allows the normal data to be overridden with custom--i.e. coadded frames.
+    # This is clunky and this function should probably allow for co-adding right here, so that
+    # it's clear what is happening, because later in this function is where we divide by number of frames.
+    if image_to_plot is not None: 
+        data = image_to_plot 
+    else: # If the image to plot isn't passed in, code will try to plot the "integration"th integration
+        try:
+            # l1b where dark has already been subtracted
+            data = myfits['detector_dark_subtracted'].data[integration]
+        except:
+            # l1a without dark subtraction
+            data = myfits['Primary'].data[integration]
+    
+    # figure out the binning: There's more than one pixel per bin, so we need to convert the binned data
+    # to data in pixels
+    spapixrange = get_pix_range(myfits, which="spatial")
+    spepixrange = get_pix_range(myfits, which="spectral")
 
     spepixwidth = spepixrange[1:]-spepixrange[:-1]
     spapixwidth = spapixrange[1:]-spapixrange[:-1]
 
     npixperbin = np.outer(spapixwidth, spepixwidth)
 
-    data = data/npixperbin
-
+    if image_to_plot is not None:
+        num_frames = len(myfits['Primary'].data)
+    else:
+        num_frames = 1
+            
+    data = data / (num_frames * npixperbin)
+    
     # figure out what norm to use
     if norm is None:
-        if prange is None:
-            prange = [0, 100]
-        if arange is None:
-            arange = [np.percentile(data, prange[0]),
-                      np.percentile(data, prange[1])]
+        # Range decisions used to be here, but now it is done outside in the primary plotting routine
+        # Scale decisions
+        vmin = force_vmin if force_vmin is not None else arange[0]
+        vmax = force_vmax if force_vmin is not None else arange[1]
+        
         if scale == "linear":
-            norm = mpl.colors.Normalize(vmin=arange[0],
-                                        vmax=arange[1])
+            norm = mpl.colors.Normalize(vmin=vmin, vmax=vmax)
```

Comment:
In future, please avoid stylistic changes that don't affect the content of the code or fix PEP8 violations--- this kind of thing makes it much harder to read diffs. The rule is to keep the style of the original code unless there are substantive changes. This happens in lots of places, not just here.